### PR TITLE
support Scala 2.12

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/util/ChillClosureCleaner.scala [hunk 1]
```diff
@@ -0,0 +1,273 @@
+// scalastyle:off header.matches
```
Comment (andrewsmartin) [inline scio-core/src/main/scala/com/spotify/scio/util/ChillClosureCleaner.scala]: Will this be fixed upstream as well? I don't see an issue tracking it anywhere

File: scripts/multijoin.py [hunk 2]
```diff
@@ -198,7 +198,7 @@ def main(out):
           def toOptions[T](xs: Iterator[T]): Iterator[Option[T]] = if (xs.isEmpty) Iterator(None) else xs.map(Option(_))
         ''').replace('  # NOQA', '').lstrip('\n')
 
-    N = 22
+    N = 11
```
Comment (andrewsmartin) [inline scripts/multijoin.py]: I forget, what was the reason for this again?

File: scio-core/src/main/scala/com/spotify/scio/util/ChillClosureCleaner.scala (hunk 1 above)
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/util/ChillClosureCleaner.scala]: I will.

File: scripts/multijoin.py (hunk 2 above)
Comment (nevillelyh) [inline scripts/multijoin.py]: #468

**Comment by nevillelyh** [comment]
DO NOT MERGE. It's not completely fix. Not working for `object.MODULE$` case.

**Pull request closed without merging**